File: PDOO/Practica3/Java/Irrgarten/src/irrgarten/main/MainP3.java
/**
 * @file GameState.java
 * @author Miguel Angel De la Vega Rodríguez
 * @description This file contains a main class to test the game
 */

package irrgarten.main;

import irrgarten.Game;
import irrgarten.controller.Controller;
import irrgarten.UI.TextUI;


public class MainP3 {
    
    public static void main (String[] args) {
        
        final int NUMPLAYERS = 2;

        TextUI view = new TextUI();
        Game game = new Game(NUMPLAYERS);
        Controller controller = new Controller(game, view);

        controller.play();
    }
}

File: PDOO/Practica4/Ruby/Irrgarten/MainP3.rb
require_relative 'Controller/controller'
require_relative 'UI/textUI'
require_relative 'dice'
require_relative 'directions'
require_relative 'game_character'
require_relative 'game_state'
require_relative 'game'
require_relative 'labyrinth'
require_relative 'orientation'
require_relative 'monster'
require_relative 'shield'
require_relative 'player'
require_relative 'weapon'

module Irrgarten
  class MainP3
    @@NUM_PLAYERS = 1 # Number of players in the game

    def self.main
      view = UI::TextUI.new
      game = Game.new(@@NUM_PLAYERS)

      controller = Control::Controller.new(game, view)
      controller.play
    end
  end

  MainP3.main
end
